homeassistant/components/sleepiq/__init__.py

```diff
@@ -61,6 +67,22 @@ def setup(hass, config):
         _LOGGER.error(message)
         return False
 
+    def handle_set_number(call):
+        bed = call.data.get(BED, "")
+        side = call.data.get(SIDE, "")
+        new_sleep_number = call.data.get(SLEEP_NUMBER, 0)
+
+        DATA.set_new_sleep_number(bed, side, new_sleep_number)
+
+    def handle_set_favorite(call):
+        bed = call.data.get(BED, "")
+        side = call.data.get(SIDE, "")
+
+        DATA.set_to_favorite_sleep_number(bed, side)
+
+    hass.services.register(DOMAIN, SERVICE_SET_SLEEP_NUMBER, handle_set_number)
```

Comment:
Please create service voluptuous schemas that validates the service data and pass the schema when registering the service. See other integrations for examples.